analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Binary.Arithmetic.cs

```diff
@@ -55,6 +56,67 @@ private static NumberConstraint CalculateMultiply(NumberConstraint left, NumberC
         return NumberConstraint.From(min, max);
     }
 
+    private static NumberConstraint CalculateDivide(NumberConstraint left, NumberConstraint right)
+    {
+        if (right.Equals(NumberConstraint.Zero))
+        {
+            return null;
+        }
+        else if (right.Min == 0)
+        {
+            right = NumberConstraint.From(1, right.Max);
+        }
+        else if (right.Max == 0)
+        {
+            right = NumberConstraint.From(right.Min, -1);
+        }
+
+        BigInteger? min, max;
+        if (left.IsPositive && right.IsPositive)
+        {
+            min = right.Max is null ? 0 : left.Min / right.Max;
+            max = left.Max / right.Min;
+        }
+        else if (left.IsNegative && right.IsNegative)
+        {
+            min = right.Min is null ? 0 : left.Max / right.Min;
+            max = left.Min / right.Max;
+        }
+        else if (left.IsPositive && right.IsNegative)
+        {
+            min = left.Max / right.Max;
+            max = right.Min is null ? 0 : left.Min / right.Min;
+        }
+        else if (left.IsNegative && right.IsPositive)
+        {
+            min = left.Min / right.Min;
+            max = right.Max is null ? 0 : left.Max / right.Max;
+        }
+        else if (right.IsPositive)
+        {
+            min = left.Min / right.Min;
+            max = left.Max / right.Min;
+        }
+        else if (right.IsNegative)
+        {
+            min = left.Max / right.Max;
+            max = left.Min / right.Max;
+        }
+        else if (left.Min is not null && left.Max is not null)
+        {
+            var absMax = BigInteger.Max(BigInteger.Abs(left.Min.Value), left.Max.Value);
```

Comment:
I feel like that's just repeating the code.

Reply:
There is a lot going on in this `else if` branch:
- both ranges are mixed and left is a finite range => left has a finite positive range and a finite negative range
- right range also includes 0, yielding exception, but we ignore that exception is not captured by the `NumericConstraint` because of reasons... I'd say because an exception is not a number.
- the right range being mixed, it may not include 1, but surely will include -1
- the integer division against 1 and -1 yield the left range unchanged or flipped
- therefore, we need to take the biggest of the two extremes of left in absolute value
- but left is mixed, so `Abs(left.Max.Value) == left.Max.Value`
- and therefore, the final expression is `BigInteger.Max(BigInteger.Abs(left.Min.Value), left.Max.Value)`

I needed to redo these 7 logical steps and take 5 minutes to make sense of line 102, even after having done this steps yesterday. So I think the line of code deserves an explanation.